src/solver/conv_direct_naive_conv_bwd.cpp

```diff
@@ -40,6 +40,19 @@ bool ConvDirectNaiveConvBwd::IsApplicable(const ConvolutionContext& ctx) const
        miopen::IsDisabled(MIOPEN_DEBUG_CONV_DIRECT_NAIVE_CONV_BWD{}))
         return false;
 
+    const auto device_name = ctx.GetStream().GetDeviceName();
```

Comment:
This is obvious duplicate of code. Please factor out the common code into a function and use is everywhere (including ConvDirectNaiveConvKernelFile()).

Reply:
OK, done in https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1323/commits/15117e6569e045f63438f668c80374777e9fac5d

Reply:
> ...(including ConvDirectNaiveConvKernelFile()).

It is sad to say, but the above is not done. Now `ConvDirectNaiveConvKernelFile()` and `ConvDirectNaiveConvIsApplicableByKernelType()` contain duplicate code.

I recommend implementing function like `ConvDirectNaiveConvIsAssemblyKernel()` and using it in both `ConvDirectNaiveConvKernelFile()` and `ConvDirectNaiveConvIsApplicableByKernelType()`.

Reply:
ok